File: my-folder/0011-container-with-most-water/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i,j=0,len(height)-1
        cap=0
        while i<j:
            cap=max(cap,(j-i)*min(height[i],height[j]))
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return cap

File: my-folder/0011-container-with-most-water/solution.java
import static java.lang.Math.max;

class Solution {
    public int maxArea(int[] height) {
        int maxarea=0;
        int l = 0;
        int r = height.length-1;
        while (l<r){
            maxarea=Math.max(Math.min(height[l],height[r])*(r-l),maxarea);
            if (height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxarea;
    }
}
